File: .cache/claude/global.anthropic.claude-opus-5~5b1m~5d/sha256-418b363075cd2c49d3b7effbd33f2a2401808596c366336802fc74c082011a9b/sha256-38bd0fa51a0996d31a21639d3e6d647cce2c9843c1aea71a4cf36f23febf306b/after/.cref/abi.c
#include <stdio.h>
struct ConfigFlags { unsigned int verbose:1, debug:1, optimize:1, cache_enabled:1, log_level:3, reserved:1; };
struct ProcessState { struct ConfigFlags flags; int base_value; int multiplier; char operation; };
int main(void){
  printf("C  ConfigFlags  size=%zu align=%zu\n", sizeof(struct ConfigFlags), _Alignof(struct ConfigFlags));
  printf("C  ProcessState size=%zu align=%zu\n", sizeof(struct ProcessState), _Alignof(struct ProcessState));
  printf("C  offsets: flags=%zu base=%zu mult=%zu op=%zu\n",
    __builtin_offsetof(struct ProcessState,flags), __builtin_offsetof(struct ProcessState,base_value),
    __builtin_offsetof(struct ProcessState,multiplier), __builtin_offsetof(struct ProcessState,operation));
  return 0; }

File: .cache/claude/global.anthropic.claude-opus-5~5b1m~5d/sha256-3b42993d00840a3fb8dd81fe5d3ce069652982fba93c20a7756c08ddf9f2f64a/before/.cref/abi.rs
#[repr(C, align(4))] struct ConfigFlags { storage: [u8;4] }
#[repr(C)] struct ProcessState { flags: ConfigFlags, base_value: i32, multiplier: i32, operation: i8 }
fn main(){
  println!("RS ConfigFlags  size={} align={}", std::mem::size_of::<ConfigFlags>(), std::mem::align_of::<ConfigFlags>());
  println!("RS ProcessState size={} align={}", std::mem::size_of::<ProcessState>(), std::mem::align_of::<ProcessState>());
  println!("RS offsets: flags={} base={} mult={} op={}",
    std::mem::offset_of!(ProcessState,flags), std::mem::offset_of!(ProcessState,base_value),
    std::mem::offset_of!(ProcessState,multiplier), std::mem::offset_of!(ProcessState,operation));
}
